src/index.js

```diff
@@ -34,34 +55,29 @@ export default function remember(source) {
           value = data
         }
 
-        sinks.slice(0).forEach(sink => {
+        sinks.slice(0).forEach((sink, index) => {
+          if (!handshook[index]) {
```

Comment:
This is not 100% foolproof/async-safe since we're not also copying the array of booleans. I guess ideally the sinks and booleans would be kept in an array together? As is, this is probably good enough, but not perfect.